spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -334,41 +317,40 @@ object Engine {
     }
   }
 
-  private def getSparkMaster : String = {
-    System.getProperty("spark.master")
+  /**
+   * Extract spark executor number and executor cores from environment.
+   * @return (nExecutor, executorCore)
+   */
+  private[utils] def sparkExecutorAndCore(): Option[(Int, Int)] = {
+    parseExecutorAndCore(SparkContext.getOrCreate().getConf)
```

Comment:
Looks good to me. Not sure why we didn't do this at the first try any other concerns? @yiheng